### ISSUE TITLE
Amazfit Cor 2 compatibility 

### ISSUE THREAD

**Issue by NicolasPopy** [Freeyourgadget/Gadgetbridge#1500]
#### Before requesting a new feature, please confirm the following:
- [ X ] I have read the [wiki](https://github.com/Freeyourgadget/Gadgetbridge/wiki), and I didn't find a solution to my problem / an answer to my question.
- [ X ] I have searched the [issues](https://github.com/Freeyourgadget/Gadgetbridge/issues), and I didn't find a solution to my problem / an answer to my question.
- [ X ] If you upload an image or other content, please make sure you have read and understood the [github policies and terms of services](https://help.github.com/articles/github-terms-of-service/#1-responsibility-for-user-generated-content)

#### Your issue is:
I have an Amazfit Cor 2 and i want to know how to help you to add compatibility

#### Your wearable device is:
- Amazfit Cor 2 (English Firmware)

**Comment by ashimokawa** [comment]
@NicolasPopy 
Casn you try to build master and try if it "just works" now?
Just be aware that we do not decide between amazfit cor and amazfit cor 2 firmware yet, so flashing might go wrong.

**Comment by dehnhard** [comment]
I also have this device and am interested in the support.

@ashimokawa The bluetooth name of my device is "Amazfit Band 2", you'll want to add that, too.

I tried it myself without luck, but I am not sure I did it right, because I have no device that works with Gadgetbridge.

**Comment by ashimokawa** [comment]
Thanks! Will also add Amazfit Band 2. After a Firmware upgrade it should become Amazfit Cor 2 though...

**Comment by dehnhard** [comment]
Really? 'Mi Fit' tells me the bands firmware V0.3.0.24 is the latest.

**Comment by ashimokawa** [comment]
@dehnhard 
I see, "Amazfit Band 2" seems to be the western version, with completely different firmware version scheme. Anyway, can you try master?

**Comment by dehnhard** [comment]
No success with master for me, but as I already wrote I am not sure about the correct way to connect it. Is it better to try while it's connected to "Mi Fit" or while disconnected? Any way to debug? E.g. a string to look for in ADB logs?

**Comment by ashimokawa** [comment]
You have to disconnect mi fit, maybe even uninstall, I dont know since I newer saw mi fit.

Then remove from paired devices in android settings. After that press the + button in Gadgetbridge and tap on the device.

You should only do tests, when we release the next version it might be that Cor2 will become a different device id internally (currently handled as Cor).

**Comment by dehnhard** [comment]
Ok, removing from paired devices did the trick, that GadgetBridge found the device, but the pairing process after tapping it, doesn't finish. The device never shows the dialogue to accept it.

**Comment by dehnhard** [comment]
Success! 
I tried again today and it worked. The only difference was that I entered a name for the device ... 
So far I can attest that the clock and notifications are working. 
What doesn't work is the feature that the device turns on when turning the wrist to look at it (I guess it's a new feature).
Please tell me if there is something specific to test.

**Comment by ashimokawa** [comment]
@dehnhard

There is an optionin Mi Band / Amazfit options

I will rework all the options stuff soon, I know it is confusing

**Comment by ashimokawa** [comment]
I am still curious what the _w variants are that pop up lately (Cor 2, Bip 2, etc). They follow a different firmware scheme and we do not know if they have different hardware.

I guess _w is for "western". Most of the time they still share the same device name, but in case of Cor it is "Amazfit Band 2" (_w) and "Amazfit Cor 2" (non _w), The original Cor was also "Amazfit Band" but became "Cor", very strange.

Does anyone know more or is willing to brick his/her Band by cross flashing? :scream:

**Comment by ashimokawa** [comment]
@dehnhard
master uses a new id for the Cor 2 now, and flashing the firmware is supported for fearless people.
You should delete and re-pair your device when you update master or install the next release.

If you have root you can fix it yourself in the sqlite database by setting the type column in the "DEVICE" table to 15 (from 13)

**Comment by dehnhard** [comment]
I did update and it's working fine after re-pairing. I noticed that the device menu (swipe down on the device) disappeared. This menu is configurable in "Amazfit Cor" and it did work with the previous code.

P.S. I'll leave the "fearless firmware flashing"(TM) to someone else, as I intend to spend my limited free time on testing functions.

**Comment by ashimokawa** [comment]
@dehnhard 
Do you mean you cannot enter any menu? Are you sure you set your user details? Weight and height and stuff?

**Comment by dehnhard** [comment]
@ashimokawa I meant there was no menu or a menu without items, just a black screen.

Entering the user details did show the menu again. Looks like a bug or usability problem to me. If it's really necessary to enter data (maybe it works with zeros and empty strings instead of null? just guessing...), then the data entry should be mandatory.

**Comment by IzzySoft** [comment]
With the Amazfit Cors 2 listed as supported device now, is there something left open in this issue – or can it be closed?

**Comment by ashimokawa** [comment]
@IzzySoft 
Yes closing this, the empty menu issue was closed in 8449e55f9e9c174a27f67a52cf0cbfce7ab8e765 long ago.
